library/dnet.c

```diff
@@ -474,13 +866,15 @@ static int dnet_iterator_callback_common(void *priv, struct dnet_raw_id *key, ui
 	response->flags = flags;
 	dnet_convert_iterator_response(response);
 
-	/* Data */
-	err = dnet_read_ll(fd, (char *)position, dsize, data_offset);
-	if (err)
-		goto err_out_exit;
+	if (dsize) {
```

Comment:
Relying on knowing internal structure of the read call is not clear. When there is explicit size check reader knows everything is safe.